src/components/chip-group/chip-group.tsx

```diff
@@ -0,0 +1,199 @@
+    this.updateItems;
+  }
+
+  componentDidRender(): void {
+    updateHostInteraction(this);
+  }
+
+  disconnectedCallback(): void {
+    this.mutationObserver?.disconnect();
+  }
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("calciteInternalChipKeyEvent")
+  calciteInternalChipKeyEvent(event: CustomEvent): void {
+    if (this.el === event.detail.target.parentElement) {
+      switch (event.detail.key) {
+        case "ArrowRight":
+          focusElementInGroup(this.items, event.detail.target, "next");
+          break;
+        case "ArrowLeft":
+          focusElementInGroup(this.items, event.detail.target, "previous");
+          break;
+        case "Home":
+          focusElementInGroup(this.items, event.detail.target, "first");
+          break;
+        case "End":
+          focusElementInGroup(this.items, event.detail.target, "last");
+          break;
+      }
+    }
+  }
+
+  @Listen("calciteChipClose")
+  calciteChipListener(event: CustomEvent): void {
+    const item = event.target as HTMLCalciteChipElement;
+    if (this.items.includes(item)) {
+      if (this.items.indexOf(item) > 0) {
+        focusElementInGroup(this.items, item as HTMLCalciteChipElement, "previous");
+      } else if (this.items.indexOf(item) === 0) {
+        focusElementInGroup(this.items, item as HTMLCalciteChipElement, "next");
+      } else {
+        focusElementInGroup(this.items, item as HTMLCalciteChipElement, "first");
+      }
+    }
+    this.items = this.items.filter((el) => el !== item);
+  }
+
+  @Listen("calciteChipSelect")
+  updateActiveItemOnChange(event: CustomEvent): void {
+    this.calciteChipInternalSelectionChange.emit(event.target as HTMLCalciteChipElement);
+    this.selectedItems = this.items.filter((item) => item.selected);
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
+
+  private updateItems = (event: Event): void => {
+    const updatedChips = (event.target as HTMLSlotElement)
+      .assignedElements({ flatten: true })
+      .filter((el) => el?.matches("calcite-chip")) as HTMLCalciteChipElement[];
+
+    updatedChips?.map((el) => {
+      el.selectable = this.selectionMode !== "none";
+      el.selectionMode = this.selectionMode;
+    });
+
+    this.items = updatedChips;
+  };
+
+  //--------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  //--------------------------------------------------------------------------
+
+  render(): VNode {
+    let role = undefined;
+    switch (this.selectionMode) {
+      case "single":
+      case "single-persist":
+        role = "radiogroup";
+        break;
```

Comment:
The role works well for `"single"` and `"single-persist"`, maybe for the other two (`"none"` and `"multiple"`) we could use the `"group"` `role`?

Something like:
```ts
    let role = "group";
    switch (this.selectionMode) {
      case "single" || "single-persist":
        role = "radiogroup";
        break;
```

Reply:
Added :)